src/Components/Web.Extensions/src/InputFile/IBrowserFile.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components.Web.Extensions
+{
+    /// <summary>
+    /// Represents the data of a file selected from an <see cref="InputFile"/> component.
+    /// </summary>
+    public interface IBrowserFile
+    {
+        /// <summary>
+        /// Gets the name of the file.
+        /// </summary>
+        string Name { get; }
+
+        /// <summary>
+        /// Gets the last modified date.
+        /// </summary>
+        DateTime LastModified { get; }
+
+        /// <summary>
+        /// Gets the size of the file in bytes.
+        /// </summary>
+        long Size { get; }
+
+        /// <summary>
+        /// Gets the MIME type of the file.
+        /// </summary>
+        string Type { get; }
+
+        /// <summary>
+        /// Opens the stream for reading the uploaded file.
+        /// </summary>
+        Stream OpenReadStream();
+
+        /// <summary>
+        /// Converts the current image file to a new one of the specified file type and maximum file dimensions.
+        /// </summary>
+        /// <remarks>
+        /// The image will be scaled to fit the specified dimensions while preserving the original aspect ratio.
+        /// </remarks>
+        /// <param name="format">The new image format.</param>
+        /// <param name="maxWith">The maximum image width.</param>
+        /// <param name="maxHeight">The maximum image height</param>
+        /// <returns>A <see cref="Task"/> representing the completion of the operation.</returns>
+        Task<IBrowserFile> ToImageFileAsync(string format, int maxWith, int maxHeight);
```

Comment:
It's not supper big/important, but is there a specific reason to have this "super specific" functionality here? This is more of a question for @pranavkm / @SteveSandersonMS.

Reply:
I agree. This feels really out of place compared to the rest. Is this because browser has an API like this?

Reply:
This function scales the image down to fit the maximum width and height by rendering to an HTML5 canvas and encoding the result in the specified format, so it's not technically a simple browser API.

I think the reasoning behind this was that it's a common scenario to want to limit the dimensions of an uploaded image, and it would be nice to scale down large images in JS before sending them over SignalR instead of having to transfer a massive file only to scale it down in .NET.

While possible, it would be difficult for users to do this because it would require knowledge of the workings of `inputFile.js` since it contains the file bookkeeping logic.

Reply:
I'm "OK-ish" with leaving this for now, but I would like us to find a better solution in the future. One possibility would be to change this function to something like `TransformAsync(<<JSObjectRef>>|<<function-identifier>>)` and make it so that when you call this, we Invoke whatever function/JSObjectRef you've given us store the results in a new IBrowserFileInstance.

That's much more agnostic and doesn't include any encode any specific/specialized behavior within this library. I'm not saying we do this right now, but we should strongly consider changing this before RTM.